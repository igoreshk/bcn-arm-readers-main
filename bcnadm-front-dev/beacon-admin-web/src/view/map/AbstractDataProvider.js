/* eslint-disable no-unused-vars */
/* @flow*/

class AbstractDataProvider {
  /**
   * @param {string} buildingId
   * @param {string} levelId
   * @param {object} dto
   * @param {object} entityId
   * @returns {Promise}
   */
  getDtos(buildingId, levelId) {
    throw Error('Method not supported');
  }

  getDto(buildingId, levelId, markerId) {
    throw Error('Method not supported');
  }

  saveDto(buildingId, levelId, dto) {
    throw Error('Method not supported');
  }

  removeDto(buildingId, levelId, dto) {
    throw Error('Method not supported');
  }

  removeAll(buildingId, levelId) {
    throw Error('Method not supported');
  }
}

export default AbstractDataProvider;
