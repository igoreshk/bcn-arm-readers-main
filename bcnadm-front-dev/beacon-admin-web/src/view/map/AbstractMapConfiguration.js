/* eslint-disable no-unused-vars */
/* @flow*/

export default class AbstractMapConfiguration {
  /**
   * createMap
   * @returns {Promise}
   */
  createMap() {
    throw Error('Method not supported');
  }

  /**
   * setMapBackground
   * @param {object} map
   * @returns {void}
   */
  setMapBackground(map) {
    throw Error('Method not supported');
  }

  /**
   * getImageAndCalculateBounds
   * @param {object} map
   * @param {string} levelId
   * @param {string} buildingId
   * @returns {Promise}
   */
  getImageAndCalculateBounds(map, levelId, buildingId) {
    throw Error('Method not supported');
  }

  /**
   * configureOverlay
   * @param {object} map
   * @param {object} bounds
   * @param {string} image
   * @returns {Promise}
   */
  configureOverlay(map, bounds, image) {
    throw Error('Method not supported');
  }

  /**
   * setMapDraggingListener
   * @param {object} map
   * @param {object} bounds
   */
  setMapDraggingListener(map, bounds) {
    throw Error('Method not supported');
  }
}
