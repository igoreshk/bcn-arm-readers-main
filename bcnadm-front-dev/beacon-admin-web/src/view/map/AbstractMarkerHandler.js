/* eslint-disable no-unused-vars */
/* @flow*/

class AbstractMarkerHandler {
  /**
   * createMarker
   * @param {object} mapProvider
   * @param {string} icon
   * @param {object} location
   */
  createMarker(mapProvider, icon, location) {
    throw Error('Method not supported');
  }

  /**
   * createPolyline
   * @param {object} mapProvider
   * @param {object} map
   * @param {array} coordinates
   */
  createPolyline(mapProvider, map, coordinates) {
    throw Error('Method not supported');
  }

  /**
   * createScalingPolyline
   * @param {object} mapProvider
   * @param {object} map
   * @param {array} coordinates
   */
  createScalingPolyline(mapProvider, map, coordinates) {
    throw Error('Method not supported');
  }

  /**
   * setEdgeSeparateListener
   * @param {object} mapProvider
   * @param {object} map
   * @param {object} edge
   * @param {function} onEdgeSeparate
   */
  setEdgeSeparateListener(mapProvider, map, edge, onEdgeSeparate) {
    throw Error('Method not supported');
  }

  /**
   * setScaleEdgeListener
   * @param {object} mapProvider
   * @param {object} edge
   * @param {function} showDialog
   */
  setScaleEdgeListener(mapProvider, edge, showDialog) {
    throw Error('Method not supported');
  }

  /**
   * setEdgeMovingListener
   * @param {object} mapProvider
   * @param {object} map
   * @param {object} edge
   * @param {function} onEdgeMoving
   */
  setEdgeMovingListener(mapProvider, map, edge, onEdgeMoving) {
    throw Error('Method not supported');
  }

  /**
   * setMapListener
   * @param {object} mapProvider
   * @param {object} map
   * @param {function} setMarker
   */
  setMapListener(mapProvider, map, setMarker) {
    throw Error('Method not supported');
  }

  /**
   * setMapRightClickListener
   * @param {object} mapProvider
   * @param {object} map
   * @param {function} stopVerticesChain
   */
  setMapRightClickListener(mapProvider, map, stopVerticesChain) {
    throw Error('Method not supported');
  }

  /**
   * setVertexClickListener
   * @param {object} mapProvider
   * @param {object} marker
   * @param {function} connectMarkers
   * @param {function} changeVertexIcon
   */
  setVertexClickListener(mapProvider, marker, connectMarkers, changeVertexIcon) {
    throw Error('Method not supported');
  }

  /**
   * setDragEndListener
   * @param {object} mapProvider
   * @param {object} marker
   * @param {function} save
   */
  setDragEndListener(mapProvider, marker, save) {
    throw Error('Method not supported');
  }

  /**
   * setDragStartListener
   * @param {object} mapProvider
   * @param {object} map
   * @param {object} marker
   * @param {function} onMarkerDragStart
   */
  setDragStartListener(mapProvider, map, marker, onMarkerDragStart) {
    throw Error('Method not supported');
  }

  /**
   * setRightClickListener
   * @param {object} mapProvider
   * @param {object} marker
   * @param {function} remove
   */
  setRightClickListener(mapProvider, marker, remove) {
    throw Error('Method not supported');
  }

  /**
   * setClickListener
   * @param {object} mapProvider
   * @param {object} marker
   * @param {function} edit
   */
  setClickListener(mapProvider, marker, edit) {
    throw Error('Method not supported');
  }

  /**
   * removeScaleEdgeListener
   * @param {object} mapProvider
   * @param {object} edge
   */
  removeScaleEdgeListener(mapProvider, edge) {
    throw Error('Method not supported');
  }

  /**
   * removeEdgeSeparateListener
   * @param {object} mapProvider
   * @param {object} edge
   */
  removeEdgeSeparateListener(mapProvider, edge) {
    throw Error('Method not supported');
  }

  /**
   * removeEdgeMovingListener
   * @param {object} mapProvider
   * @param {object} edge
   */
  removeEdgeMovingListener(mapProvider, edge) {
    throw Error('Method not supported');
  }

  /**
   * removeMapListener
   * @param {object} mapProvider
   * @param {object} map
   */
  removeMapListener(mapProvider, map) {
    throw Error('Method not supported');
  }

  /**
   * removeDragEndListener
   * @param {object} mapProvider
   * @param {object} marker
   */
  removeDragEndListener(mapProvider, marker) {
    throw Error('Method not supported');
  }

  /**
   * removeDragStartListener
   * @param {object} mapProvider
   * @param {object} marker
   */
  removeDragStartListener(mapProvider, marker) {
    throw Error('Method not supported');
  }

  /**
   * removeRightClickListener
   * @param {object} mapProvider
   * @param {object} marker
   */
  removeRightClickListener(mapProvider, marker) {
    throw Error('Method not supported');
  }

  /**
   * removeClickListener
   * @param {object} mapProvider
   * @param {object} marker
   */
  removeClickListener(mapProvider, marker) {
    throw Error('Method not supported');
  }

  /**
   * removeVertexClickListener
   * @param {object} mapProvider
   * @param {object} marker
   */
  removeVertexClickListener(mapProvider, marker) {
    throw Error('Method not supported');
  }

  /**
   * setMarkerOnMap
   * @param {object} marker
   * @param {object} map
   */
  setMarkerOnMap(marker, map) {
    throw Error('Method not supported');
  }

  /**
   * setEdgeOnMap
   * @param {object} edge
   * @param {object} map
   */
  setEdgeOnMap(edge, map) {
    throw Error('Method not supported');
  }

  /**
   * removeMarkerFromMap
   * @param {object} marker
   */
  removeMarkerFromMap(marker) {
    throw Error('Method not supported');
  }

  /**
   * removeEdgeFromMap
   * @param {object} edge
   */
  removeEdgeFromMap(edge) {
    throw Error('Method not supported');
  }

  /**
   * setDraggable
   * @param {object} marker
   */
  setDraggable(marker) {
    throw Error('Method not supported');
  }

  /**
   * setUndraggable
   * @param {object} marker
   */
  setUndraggable(marker) {
    throw Error('Method not supported');
  }

  /**
   * setInfoTooltip
   * @param {object} mapProvider
   * @param {object} marker
   * @param {object} map
   * @param {promise} info
   */
  setInfoTooltip(mapProvider, marker, map, info) {
    throw Error('Method not supported');
  }

  /**
   * setEdgeInfoTooltip
   * @param {object} mapProvider
   * @param {object} edge
   * @param {object} map
   * @param {promise} info
   */
  setEdgeInfoTooltip(mapProvider, edge, map, info) {
    throw Error('Method not supported');
  }
}

export default AbstractMarkerHandler;
