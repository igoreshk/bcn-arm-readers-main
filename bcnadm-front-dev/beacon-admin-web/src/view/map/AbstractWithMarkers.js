/* eslint-disable no-unused-vars */
/* @flow*/
import { Component } from 'react';

class AbstractWithMarkers extends Component {
  /**
   * @param {object} marker
   * @returns {void}
   */
  onMarkerSelect(marker) {
    throw Error('This is the abstract class');
  }

  /**
   * @returns {Promise}
   */
  onMarkerRemove(marker) {
    throw Error('This is the abstract class');
  }

  onMarkerDragStart() {
    throw Error('This is the abstract class');
  }

  onMarkerDrop() {
    throw Error('This is the abstract class');
  }

  /**
   * @returns {Promise}
   */
  onMarkerDragEnd(marker) {
    throw Error('This is the abstract class');
  }

  /**
   * @returns {void}
   */
  onMarkerCreate(location) {
    throw Error('This is the abstract class');
  }

  /**
   * @param {String} levelId
   * @returns {Promise}
   */
  onRemoveAllMarkers(levelId) {
    throw Error('This is the abstract class');
  }

  /**
   * @returns {String}
   */
  infoTooltip(marker) {
    throw Error('This is the abstract class');
  }

  changeEditMode() {
    throw Error('This is the abstract class');
  }
}

export default AbstractWithMarkers;
