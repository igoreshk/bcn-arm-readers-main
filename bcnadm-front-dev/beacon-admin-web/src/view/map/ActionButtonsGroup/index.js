import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import Edit from '@material-ui/icons/Edit';
import PropTypes from 'prop-types';

import ModeButton, { buttonType } from '../ModeButton';
import ConfirmDeleteDialog from '../ConfirmDeleteDialog';

import './actionButtonsGroup.scss';

class ActionButtonsGroup extends Component {
  render() {
    const { clear, translate, toEdit } = this.props;
    const { edit: editMode } = this.props.match.params;

    function setClassName() {
      return editMode ? 'activeEditButtonIcon' : 'inactiveEditButtonIcon';
    }

    return (
      <div className="actionButtonsGroup">
        <ModeButton
          icon={<Edit className={setClassName()} />}
          onClick={toEdit}
          active={Boolean(editMode)}
          tooltip={translate('operator.edit')}
          buttonId="edit"
          buttonType={buttonType.ACTION}
        />
        {editMode && <ConfirmDeleteDialog clear={clear} translate={translate} />}
      </div>
    );
  }
}

ActionButtonsGroup.propTypes = {
  translate: PropTypes.func.isRequired,
  clear: PropTypes.func.isRequired,
  match: PropTypes.object,
  toEdit: PropTypes.func
};

export default withRouter(ActionButtonsGroup);
