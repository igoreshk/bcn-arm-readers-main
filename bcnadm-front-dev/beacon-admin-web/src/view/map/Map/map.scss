.mapWrapper {
  flex: 1 0;
  height: calc(100vh - 56px);
  z-index: 1;
}
.mapContainer {
  height: 100%;
}
