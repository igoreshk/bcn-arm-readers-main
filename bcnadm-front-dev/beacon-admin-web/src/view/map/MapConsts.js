/**
 * Contains set of constants for configuring map
 * @namespace mapConsts
 * @prop {object} mapOptions Constant, used in google.maps.Map() constructor when creating map
 * @prop {function} configureCoordMapType Configures overlay map type & map tiles
 * (See: {@link configureCoordMapType})
 * @prop {function} configureOverlay Configures overlay, particularly sets custom image as an overlay, size of an overlay etc
 * (See: {@link configureOverlay})
 *
 * @see [Custom overlays]{@link https://developers.google.com/maps/documentation/javascript/customoverlays?hl=en}
 * To read about configuring custom overlays
 */
export const INITIAL_ZOOM = 18;

const mapConsts = {
  mapOptions: {
    zoom: INITIAL_ZOOM,
    minZoom: 12,
    maxZoom: 28,
    center: { lat: 0, lng: 0 },
    mapTypeId: 'roadmap',
    zoomControl: true,
    mapTypeControl: false,
    scaleControl: true,
    streetViewControl: false,
    rotateControl: false,
    fullscreenControl: false
  },
  /**
   * This function configures tiles of map
   * @function
   * @param coord {object} Coordinates on current tile
   * @param zoom {number} Zoom rate
   * @param ownerDocument {object} returns higher-level document object for current node
   * @returns {object} configured HTML element
   */
  divConfig(coord, zoom, ownerDocument) {
    const div = ownerDocument.createElement('div');
    div.innerHTML = coord;
    div.style.width = `${this.tileSize.width}px`;
    div.style.height = `${this.tileSize.height}px`;
    div.style.fontSize = '0';
    div.style.borderStyle = 'solid';
    div.style.borderWidth = '10000px';
    div.style.borderColor = '#e5e5e5';
    return div;
  },
  /**
   * This function is called when the map is ready for the overlay to be attached
   *@function
   */
  onAdd() {
    const div = document.createElement('div');
    div.style.border = 'none';
    div.style.borderWidth = '0px';
    div.style.position = 'absolute';

    const img = document.createElement('img');
    img.src = this.image;
    img.style.width = '100%';
    img.style.height = '100%';
    div.appendChild(img);

    this.div = div;
    const panes = this.getPanes();
    panes.overlayLayer.appendChild(div);
  },
  /**
   * This function  will be called when the object (custom overlay) is first displayed
   * @function
   */
  draw() {
    const overlayProjection = this.getProjection();
    const sw = overlayProjection.fromLatLngToDivPixel(this.bounds.getSouthWest());
    const ne = overlayProjection.fromLatLngToDivPixel(this.bounds.getNorthEast());
    const div = this.div;
    this.div.id = 'layer';
    div.style.left = `${sw.x}px`;
    div.style.top = `${ne.y}px`;
    div.style.width = `${ne.x - sw.x}px`;
    div.style.height = `${sw.y - ne.y}px`;
    div.style.transform = `rotate(${this.rotation}rad)`;
  },
  /**
   * This function is used to clean up any elements added within the overlay.
   * @function
   */
  onRemove() {
    this.div.parentNode.removeChild(this.div);
    this.div = null;
  }
};
export const TILE_SIZE = 1000;
export default mapConsts;
/**
 * Mode when user adds beacon markers
 * @constant
 * @type {string}
 */
const BEACON_MODE = 'beacons';
/**
 * Mode when user adds reader markers
 * @constant
 * @type {string}
 */
const READER_MODE = 'readers';
/**
 * Mode when user adds vertex markers
 * @constant
 * @type {string}
 */
const ROUTE_MODE = 'routing';
/**
 * Mode when user adds area markers
 * @constant
 * @type {string}
 */
const AREAS_MODE = 'areas';
/**
 * Mode when user monitore people on the map
 * @constant
 * @type {string}
 */
const MONITORING_MODE = 'monitoring';
/**
 * Mode when user is able to draw on the map tile
 * @constant
 * @type {string}
 */
const DRAWING_MODE = 'drawing';
/**
 * Represents state of a dialog, where beacon's info can be configures
 * @constant
 * @type {string}
 */
const BEACON_DIALOG_OPEN = 'BEACON_DIALOG_OPEN';
/**
 * Represents state of a dialog, where beacon's info can be configures
 * @constant
 * @type {string}
 */
const BEACON_DIALOG_CLOSE = 'BEACON_DIALOG_CLOSE';
/**
 * Represents state of a dialog, where area's info can be configures
 * @constant
 * @type {string}
 */
const AREAS_DIALOG_OPEN = 'AREAS_DIALOG_OPEN';

const READER_DIALOG_OPEN = 'READER_DIALOG_OPEN';

const READER_DIALOG_CLOSE = 'READER_DIALOG_CLOSE';

const DISTANCE_DIALOG_OPEN = 'DISTANCE_DIALOG_OPEN';

const DISTANCE_DIALOG_CLOSE = 'DISTANCE_DIALOG_CLOSE';

/**
 * Represents state of a dialog, where area's info can be configures
 * @constant
 * @type {string}
 */
const AREAS_DIALOG_CLOSE = 'AREAS_DIALOG_CLOSE';

/**
 * Describes different modes
 * @type {{BEACON_MODE: string, ROUTE_MODE: string, AREAS_MODE: string}}
 */
const SCALE_MODE = 'scaling';

const GOOGLE = 'Google';

const OSM = 'OSM';

export const modes = {
  READER_MODE,
  BEACON_MODE,
  ROUTE_MODE,
  AREAS_MODE,
  SCALE_MODE,
  MONITORING_MODE,
  DRAWING_MODE
};
/**
 * Set of area/beacon dialog states
 * @constant
 * @type {object}
 */
export const dialog = {
  BEACON_DIALOG_OPEN,
  BEACON_DIALOG_CLOSE,
  AREAS_DIALOG_OPEN,
  AREAS_DIALOG_CLOSE,
  DISTANCE_DIALOG_OPEN,
  DISTANCE_DIALOG_CLOSE,
  READER_DIALOG_OPEN,
  READER_DIALOG_CLOSE
};

export const mapProviders = {
  GOOGLE,
  OSM
};
