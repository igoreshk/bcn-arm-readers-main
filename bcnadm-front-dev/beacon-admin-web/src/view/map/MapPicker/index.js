import React from 'react';
import { withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { Select, MenuItem, FormControl } from '@material-ui/core';
import { setMapProviderName } from 'src/reducers/mapSettingsReducer/mapSettingsSlice';
import PropTypes from 'prop-types';

import './mapPicker.scss';

const MapPicker = ({ selectedMapProviderName, setProviderName }) => {
  const handleMapChange = (event) => {
    const newMap = event.target.value;
    if (newMap !== selectedMapProviderName) {
      setProviderName({ selectedMapProviderName: newMap });
    }
  };

  return (
    <FormControl className="filledForm">
      <Select value={selectedMapProviderName} onChange={handleMapChange} displayEmpty>
        <MenuItem value="Google">Google</MenuItem>
        <MenuItem value="OSM">OSM</MenuItem>
      </Select>
    </FormControl>
  );
};

const mapStateToProps = (state) => {
  return {
    selectedMapProviderName: state.mapSettings.selectedMapProviderName
  };
};

const mapDispatchToProps = {
  setProviderName: setMapProviderName
};

MapPicker.propTypes = {
  selectedMapProviderName: PropTypes.string.isRequired,
  setProviderName: PropTypes.func
};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(MapPicker));
