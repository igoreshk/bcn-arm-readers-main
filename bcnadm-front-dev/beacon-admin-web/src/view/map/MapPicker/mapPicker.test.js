import React from 'react';
import { shallow } from 'enzyme';
import { mapProviders } from 'src/view/map/MapConsts';

import MapPicker from './index';

const props = {
  selectedMap: mapProviders.GOOGLE,
  setMap: jest.fn()
};

const wrapper = shallow(<MapPicker {...props} />);

describe('mock', () => {
  it('render component', () => expect(wrapper.length).toEqual(1));
  it('snapshot for MapPicker', () => expect(wrapper).toMatchSnapshot());
});
