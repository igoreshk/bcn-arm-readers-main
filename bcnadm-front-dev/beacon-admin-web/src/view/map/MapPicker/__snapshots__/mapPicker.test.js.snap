// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mock snapshot for MapPicker 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
